### PR TITLE
Add Time Series Workflow

### PR DESCRIPTION
**Pull request by AndrewPlayer3**
Confirmed that output still matches after changes.

### CODE REVIEW

File: src/hyp3_srg/time_series.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+        if not geo_name.exists():
+            shutil.unpack_archive(name.with_suffix('.zip'), work_dir)
+
+        granule_names.append(str(name))
+
+    return granule_names
+
+
+def get_size_from_dem(dem_file: str) -> tuple[int]:
```
Comment (forrestfwilliams) [inline src/hyp3_srg/time_series.py:74]: Could use a test to ensure the read is happening correctly.

File: src/hyp3_srg/time_series.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    utils.call_stanford_module('util/unwrap_parallel.py', args=[unw_width], work_dir=work_dir)
+
+
+def compute_sbas_velocity_solution(
+    threshold: float, do_tropo_correction: bool, unw_shape: tuple[int], work_dir: Path
+) -> None:
+    """Computes the sbas velocity solution from the unwrapped interferograms
+
+    Args:
+        threshold: ...
```
Comment (forrestfwilliams) [inline src/hyp3_srg/time_series.py:135]: Can you fill out these arg lines?

File: src/hyp3_srg/time_series.py [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+
+    ref_point_args = ['unwlist', unw_width, unw_length, threshold]
+    utils.call_stanford_module('int/findrefpoints', args=ref_point_args, work_dir=work_dir)
+
+    if do_tropo_correction:
+        tropo_correct_args = ['unwlist', unw_width, unw_length]
+        utils.call_stanford_module('int/tropocorrect.py', args=tropo_correct_args, work_dir=work_dir)
+
+    sbas_velocity_args = ['unwlist', num_unw_files, num_slcs, unw_width, 'ref_locs']
+    utils.call_stanford_module('sbas/sbas', args=sbas_velocity_args, work_dir=work_dir)
```
Comment (forrestfwilliams) [inline src/hyp3_srg/time_series.py:161]:
```suggestion
    ref_point_args = ['unwlist', unw_width, unw_length, threshold]
    utils.call_stanford_module('int/findrefpoints', args=ref_point_args, work_dir=work_dir)

    if do_tropo_correction:
        tropo_correct_args = ['unwlist', unw_width, unw_length]
        utils.call_stanford_module('int/tropocorrect.py', args=tropo_correct_args, work_dir=work_dir)

    num_unw_files = 0
    with open(work_dir / 'unwlist', 'r') as unw_list:
        num_unw_files = len(unw_list.readlines())

    num_slcs = 0
    with open(work_dir / 'geolist', 'r') as slc_list:
        num_slcs = len(slc_list.readlines())
 
    sbas_velocity_args = ['unwlist', num_unw_files, num_slcs, unw_width, 'ref_locs']
    utils.call_stanford_module('sbas/sbas', args=sbas_velocity_args, work_dir=work_dir)
```
I think it's cleaner written like this. We don't use `num_unw_files` and `num_slcs` until the sbas call.

File: src/hyp3_srg/time_series.py [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    threshold: float = 0.5,
+    do_tropo_correction: bool = True,
+    work_dir: Path | None = None,
+) -> None:
+    """Creates a time series from a stack of GSLCs consisting of interferograms and a velocity solution
+
+    Args:
+        looks: tuple containing the number range looks and azimuth looks
+        baselines: tuple containing the time baseline and spatial baseline
+        threshold: ...
```
Comment (forrestfwilliams) [inline src/hyp3_srg/time_series.py:176]: Fill out docstring.

File: src/hyp3_srg/time_series.py [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+
+    unw_shape = get_size_from_dem(work_dir / 'dem.rsc')
+    unwrap_interferograms(dem_shape=dem_shape, unw_shape=unw_shape, work_dir=work_dir)
+
+    compute_sbas_velocity_solution(
+        threshold=threshold, do_tropo_correction=do_tropo_correction, unw_shape=unw_shape, work_dir=work_dir
+    )
+
+
+def create_time_series_product_name(
```
Comment (forrestfwilliams) [inline src/hyp3_srg/time_series.py:191]: Could use some testing.

**Review by forrestfwilliams — COMMENTED** [review]
Core functionality is great. Just some cleaning up and testing to do.

File: README.md [hunk 6]
```diff
@@ -34,7 +38,21 @@ docker run -it --rm \
     S1A_IW_RAW__0SDV_20231229T134404_20231229T134436_051870_064437_5F38-RAW
 ```
 
-## Earthdata Login
+### Time-series
+The `time_series` workflow takes a list of up to 50 Level-0 S1 granule names and produces a time-series of the respective geocoded SLCs. Stacks are created with looks `10x10`and baselines of `1000x1000`. A  trophospheric correction is applied using an elevation-dependent regression.
```
[the remark this answers is not in the document]
Comment (AndrewPlayer3) [reply README.md:42]: 10 by 10 looks is the default in SRG. Something like 4 looks are used for finding reference points (not an optional parameter). The baselines are temporal and spatial baselines - same as the defaults in SRG. I think I'll write this as:

"The `time_series` workflow takes a list of up to 50 Sentinel-1 GSLC granule names, along with a bounding box, and produces a time-series. **Note that all of the input GSLSs must have been generated with the provided bounding box.**  Stacks are created with 10 range looks, 10 azimuth looks,  and temporal and spatial baselines of 1000 and 1000, respectively. Candidate reference points are chosen with a correlation threshold of `0.5` - meaning the correlation must be above 0.5 in all scenes at that point. A tropospheric correction is applied using an elevation-dependent regression."

File: README.md [hunk 7]
```diff
@@ -6,9 +6,10 @@ HyP3 plugin for Stanford Radar Group (SRG) SAR Processor
...
 
-The HyP3-SRG plugin provides a set of workflows (currently only accessible via the docker container) that can be used to process SAR data using the [Stanford Radar Group Processor](https://github.com/asfhyp3/srg). The workflows currently included in this plugin are:
+The HyP3-SRG plugin provides a set of workflows (currently only accessible via the docker container) that can be used to process SAR data using the [Stanford Radar Group Processor](https://github.com/asfhyp3/srg). This set of workflow uses the [SRG alogorithms]((https://doi.org/10.1109/LGRS.2017.2753580)) to process Level-0 Sentinel-1 (S1) data to geocoded, user-friendly products that can be used for time-series analysis. The workflows currently included in this plugin are:
 
-- `back_projection`: A workflow for creating geocoded Sentinel-1 SLCs from Level-0 data using the [back-projection methodology](https://doi.org/10.1109/LGRS.2017.2753580).
+- [`back_projection`](#back-projection): A workflow for creating geocoded Sentinel-1 SLCs,
```
Comment (forrestfwilliams) [inline README.md:11]:
```suggestion
- [`back_projection`](#back-projection): A workflow for creating geocoded Sentinel-1 SLCs.
```

**Review by forrestfwilliams — APPROVED** [review]

**Pull request merged**